models/models_methods.go

```diff
@@ -320,31 +341,33 @@ func (workflow *WorkflowModel) Validate(secrets []envmanModels.EnvironmentItemMo
 			}
 
 			if !*isExpand {
-				return warnings, fmt.Errorf("is_sensitive option set to true but is_expand is not, sensitive inputs cannot have direct values and to be able to use environment variable for input: (%s) you need to enable is_expand", key)
+				return nil, fmt.Errorf("is_sensitive option set to true but is_expand is not, sensitive inputs cannot have direct values and to be able to use environment variable for input: (%s:%s) you need to enable is_expand", stepID, key)
 			}
 
-			if err := isSecretEnv(value, secrets); err != nil {
-				return warnings, fmt.Errorf("invalid sensitive input value for (%s): %s", key, err)
+			if !isSecretEnv(value, secrets) {
+				return nil, fmt.Errorf("invalid sensitive input value for (%s:%s): value is not a secret environment variable", key, stepID)
 			}
 		}
-		stepListItem[stepID] = step
 	}
-	return warnings, nil
+	return nil, nil
 }
 
-func isSecretEnv(value string, secrets []envmanModels.EnvironmentItemModel) error {
+func isSecretEnv(value string, secrets []envmanModels.EnvironmentItemModel) bool {
```

Comment:
i would go by returning an error instead of bool.
in case of bool, you always say that the input is not a secret, but what if the secret is invalid?